### PR TITLE
Refactor alert transports to classes

### PR DESCRIPTION
**Pull request by mcq8**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

Only moved the code to classes to have less ev[a|i]l code.
Next step could be to have the $obj as an actual object so integrators know what to expect from the var.
I'm not sure about the function name call. If you have any suggestions let me know.

### CODE REVIEW

**Comment by mcq8** [comment]
Seems like the Slack alerting is already making the project incompatible with php < 5.4.
I'll move it to the old array style and fix the linting.

**Comment by laf** [comment]
Running on my install for testing.

**Comment by laf** [comment]
Testing on mail transport works. Will code review soon.

**Comment by laf** [comment]
LGTM.

@librenms/reviewers

File: LibreNMS/Interfaces/Alert/Transport.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @copyright  2017 Tony Murray
```
Comment (murrant) [inline LibreNMS/Interfaces/Alert/Transport.php]: Wrong copyright here, pretty sure I didn't right it :)

File: LibreNMS/Interfaces/Alert/Transport.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ */
+
+namespace LibreNMS\Interfaces\Alert;
+
+interface Transport
+{
+    /**
+     * Gets called when an alert is sent
+     *
+     * @param $obj array An array created by DescribeAlert
```
Comment (murrant) [inline LibreNMS/Interfaces/Alert/Transport.php]: Can you rename this from `$obj` to `$alert_data`, or something like that.

You don't have to update all the transports.

File: LibreNMS/Interfaces/Alert/Transport.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+     * @param $opts array|true The options from $config['alert']['transports'][$transport]
+     * @return bool Returns if the call was successful
+     */
+    public function call($obj, $opts);
```
Comment (murrant) [inline LibreNMS/Interfaces/Alert/Transport.php]: Not sure I like the function name here.
Options I can think of:
submitAlert()
deliverAlert()
sendMessage()

File: LibreNMS/Interfaces/Alert/Transport.php (hunk 2 above)
Comment (murrant) [reply LibreNMS/Interfaces/Alert/Transport.php]: It would also be great if you documented the fields of `$obj` here

**Comment by murrant** [comment]
Great work @mcq8 Just a few cosmetic suggestions.

File: LibreNMS/Interfaces/Alert/Transport.php (hunk 2 above)
Comment (mcq8) [reply LibreNMS/Interfaces/Alert/Transport.php]: I would rather do the documentation trough making an object out of the alert, but I'll do that trough a separate pull request.

**Comment by InsaneSplash** [comment]
I didn't test the old php version, but moving to php 7.1 resolved the problem.

**Pull request merged**